src/middlewared/middlewared/plugins/system.py

```diff
@@ -116,6 +120,23 @@ class Config:
                         f'{schema}.syslogserver',
                         'Port must be in the range of 0 to 65535.'
                     )
+
+        if data['syslog_transport'] == 'TLS':
+            try:
+                await self.middleware.call(
+                    'certificate.query',
+                    [
+                        ('id', '=', data['syslog_tls_certificate']),
+                        ('revoked', '=', False),
+                    ],
+                    {'get': True}
+                )
+            except IndexError:
+                verrors.add(
```

Comment:
I think it might be better to use `certificate.cert_services_validation` method to check for this. That probably does not yet check for certificate's revocation status, that check can be added in it and then we should use it as it works nicely to impose certain security limits on certs being used for different services.